src/main/java/betterquesting/network/handlers/NetStationEdit.java

```diff
@@ -74,7 +76,18 @@ private static void onServer(Tuple2<NBTTagCompound, EntityPlayerMP> message)
                     UUID QID = QuestingAPI.getQuestingUUID(message.getSecond());
                     IQuest quest = QuestDatabase.INSTANCE.get(NBTConverter.UuidValueType.QUEST.readId(data));
                     ITask task = quest == null ? null : quest.getTasks().getValue(data.getInteger("taskID"));
-                    if(quest != null && task != null) oss.setupTask(QID, quest, task);
+                    if(quest != null && task != null)
+                    {
+                        if (!quest.isUnlocked(QID) && !task.isComplete(QID))
+                        {
+                            BetterQuesting.logger.warn(
+                                    MarkerManager.getMarker("SuspiciousPackets"),
+                                    "Player {} tried to set task to completed or not yet unlocked one.",
```

Comment:
This log message doesn't match up the condition (!unlocked && !complete) != (!unlocked || complete)